src/Symfony/Component/Form/Tests/Extension/Core/Type/CheckboxTypeTest.php

```diff
@@ -125,15 +125,15 @@ public function testSubmitWithEmptyValueAndFalseUnchecked()
         $this->assertNull($form->getViewData());
     }
 
-    public function testSubmitWithEmptyValueAndTrueChecked()
+    public function testSubmitWithEmptyValueAndTrueUnchecked()
     {
         $form = $this->factory->create('checkbox', null, array(
             'value' => '',
         ));
         $form->submit(true);
 
-        $this->assertTrue($form->getData());
-        $this->assertSame('', $form->getViewData());
+        $this->assertNull($form->getData());
```

Comment:
hmm, that seems kind of impossible from the `DataTransformer` due to the string-cast at `Symfony\Component\Form\Form::submit()` in line 543:

```php
// Treat false as NULL to support binding false to checkboxes.
// Don't convert NULL to a string here in order to determine later
// whether an empty value has been submitted or whether no value has
// been submitted at all. This is important for processing checkboxes
// and radio buttons with empty values.
if (false === $submittedData) {
    $submittedData = null;
} elseif (is_scalar($submittedData)) {
    $submittedData = (string) $submittedData;
}
```

The transformer will only get "1". We could handle "1" as `true` to accomplish that ... ?